venvUtils/ensureVenv.py

```diff
@@ -0,0 +1,129 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+
+import sys
+import os
+import subprocess
+import shutil
+from typing import Set
+
+"""
+A script to ensure that the NVDA build system's Python virtual environment is created and up to date.
+"""
+
+top_dir: str = os.path.join(os.path.abspath(os.path.dirname(__file__)), "..")
+venv_path: str = os.path.join(top_dir, ".venv")
+requirements_path: str = os.path.join(top_dir, "requirements.txt")
+venv_orig_requirements_path: str = os.path.join(venv_path, "_requirements.txt")
+venv_python_version_path: str = os.path.join(venv_path, "python_version")
+
+def askYesNoQuestion(message: str) -> bool:
+	"""
+	Displays the given message to the user and accepts y or n as input.
+	Any other input causes the question to be asked again.
+	@returns: True for y and n for False.
+	"""
+	while True:
+		answer = input(
+			 message + " [y/n]: "
+		) 
+		if answer == 'n':
+			return False
+		elif answer == 'y':
+			return True
+		else:
+			continue # ask again
+
+def fetchRequirementsSet(path: str) -> Set[str]:
+	"""
+	Fetches all the package lines from a pip requirements.txt file
+	returning them as a set of strings.
+	The returned set could be compared with a set from another file which would allow easy identification of which requirements were added or removed.
+	"""
+	with open(path, "r") as f:
+		lines = [x.strip() for x in f.readlines()]
+		lines = [x for x in lines if x and not x.isspace() and not x.startswith('#')]
+	return set(lines)
+
+def createVenvAndPopulate():
+	"""
+	Creates the NVDA build system's Python virtual environment and installs all required packages.
+	this function will overwrite any existing virtual environment found at c{venv_path}.
+	"""
+	print("Creating virtual environment...", flush=True)
+	subprocess.run(
+		[
+			"py", "-3.8-32",

```

Comment:
Specifying Python version explicitly in more places creates more places in which we would need to update it in the future. You can just use `sys.executable` here.